python/lsst/meas/algorithms/matcherSourceSelector.py

```diff
@@ -162,4 +162,71 @@ def _isUsable(self, source):
             and not source.get(self.fluxFlagKey) \
             and self._goodSN(source)
 
+
+class MatcherPessimisticSourceSelectorTask(MatcherSourceSelectorTask):
+    """
+    !Select sources that are useful for matching.
+
+    Good matching sources have high signal/noise, are non-blended. They need not
+    be PSF sources, just have reliable centroids.
```

Comment:
Please rewrite the brief and full description to emphasize in what way this matcher differs from `MatcherSourceSelectorTask`.

Was it necessary to write a new selector instead of improving the old? Did you check to see if your updates harmed or helped WCS fitting using the old code? I can understand if you didn't want to go to that effort, but it seems quite likely your changes would help existing WCS fitting, and if so, it would be a win for code robustness and simplicity to improve the old task.